.github/workflows/create_llvm.yml

```diff
@@ -35,23 +53,61 @@ jobs:
         with:
           path:
             llvm_install/**
-          key: llvm-${{ matrix.os }}-v${{ matrix.version }}-${{ matrix.build_type }}
+          key: llvm-${{ matrix.os }}-${{ matrix.arch }}-v${{ matrix.version }}-${{ matrix.build_type }}
+
+      - name: show arch flags
+        if: steps.cache.outputs.cache-hit != 'true'      
+        run:
+          echo ${{ matrix.arch_flags }}
 
-      - name: Checkout repo
+      - name: Checkout repo llvm
         if: steps.cache.outputs.cache-hit != 'true'
         uses: actions/checkout@v4
         with:
           repository: llvm/llvm-project
           ref: release/${{matrix.version}}.x
 
+      - name: Checkout repo ock platform
+        if: ${{ steps.cache.outputs.cache-hit != 'true' && matrix.arch != 'x86_64' }}
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: |
+            platform
+          path: ock          
+
       - name: Install Ninja
         if: steps.cache.outputs.cache-hit != 'true'
         uses: llvm/actions/install-ninja@main
 
+      - name: install aarch64 build tools
+        if: ${{ steps.cache.outputs.cache-hit != 'true' && matrix.arch == 'aarch64' }}
+        run:
+          sudo apt-get install --yes gcc-11-aarch64-linux-gnu g++-11-aarch64-linux-gnu
+
       - name: Flags checker
         if: steps.cache.outputs.cache-hit != 'true'
         run:
-          echo Building on "${{ matrix.os }}" with os_flags "${{ matrix.os_flags}}" extra flags "${{ matrix.build_type_flags}}" and  build_type "${{matrix.build_type}}"
+          echo Building on "${{ matrix.os }}" with os_flags "${{ matrix.os_flags}}" aarch "${{ matrix.arch }}"extra flags "${{ matrix.build_type_flags}}" and  build_type "${{matrix.build_type}}"
+
+      - name: cmake native for cross
```

Comment:
This name confuses me.

Reply:
okay, again i was following the documentation we have which says this is required.
Does it do it automatically?

Reply:
I believe so, yes, this is conditional on `LLVM_USE_HOST_TOOLS` which is always set to `ON` when cross-compiling.